### PR TITLE
Testnet deployment files + Renamed UpdaterRSK to BlindexUpdater and removed unused conditions based on the network

### PR DESCRIPTION
**Pull request by kazazor**
## What's the purpose of your PR?

- [ ] Fixing a bug
- [ ] A new Feature
- [X] Techdebt
- [ ] General Task

## What is your PR all about?
* Prepared a standard structure to write deployment scripts
* Added a helper function to print and wait on transactions (so we'll also show the transaction hash when deploying and not just wait in an empty terminal)
* Renamed UpdaterRSK to BlindexUpdater
* Removed hardcoded conditions to check for network-specific settings when not needed (so it will support multichain).
* Prepare the initial deployment scripts on arbitrum testnet
* Removed TODOs we don't need anymore

## Any special instructions for the reviewer?

The BDStables contracts are missing here for now. It will be done in a separate task due to the size of this PR and the fact I would still like to decide how to approach it (use the regular stables or just "dumb" ERC 20 to be used with the multichain.org interface demands.

### CODE REVIEW

File: deploy-scripts/arbitrum-test/1-bdx-shares.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types";
```
Comment (tkeidar) [inline deploy-scripts/arbitrum-test/1-bdx-shares.ts:1]: Add mint BDX script

Comment (tkeidar) [inline deploy-scripts/arbitrum-test/1-bdx-shares.ts:1]: Add mint BDX script

File: deploy-scripts/arbitrum-test/4-blindex-updater.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      args: [constants.botAddress[hre.network.name]]
+    });
+
+    console.log("Deployed BlindexUpdater to: " + updater.address);
+  });
+};
+
+func.id = __filename;
+func.tags = ["BlindexUpdater"];
+func.dependencies = ["StakingRewardsDistribution"];
```
Comment (tkeidar) [inline deploy-scripts/arbitrum-test/4-blindex-updater.ts:24]: this script is not really dependent on `StakingRewardsDistribution`, I think we can remove it

File: deploy-scripts/rsk/5_2_staking_rewards_distribution.ts [hunk 3]
```diff
@@ -21,8 +22,7 @@ async function feedStakeRewardsDistribution(hre: HardhatRuntimeEnvironment) {
 const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
   const bdx = await getBdx(hre);
   const vesting = await getVesting(hre);
-  const treasuryAddress =
-    hre.network.name.toLocaleLowerCase() === "rsk" ? constants.multisigTreasuryAddress[hre.network.name] : (await getTreasury(hre)).address;
+  const treasuryAddress = hre.network.name === "mainnetFork" ? (await getTreasury(hre)).address : constants.treasuryAddresses[hre.network.name];
```
Comment (tkeidar) [inline deploy-scripts/rsk/5_2_staking_rewards_distribution.ts:25]: We do this code more than once, lets create a getTreauryAddress function in the DeployedContractsHelper

**Review by tkeidar — CHANGES_REQUESTED** [review]
looks great 👍

File: deploy-scripts/arbitrum-test/4-blindex-updater.ts (hunk 2 above)
Comment (kazazor) [reply deploy-scripts/arbitrum-test/4-blindex-updater.ts:24]: Wouldn't that cause different order between runs when I'll add another script? And then a change in addresses...

File: deploy-scripts/rsk/5_2_staking_rewards_distribution.ts (hunk 3 above)
Comment (kazazor) [reply deploy-scripts/rsk/5_2_staking_rewards_distribution.ts:25]: Sababa

File: deploy-scripts/arbitrum-test/4-blindex-updater.ts (hunk 2 above)
Comment (tkeidar) [reply deploy-scripts/arbitrum-test/4-blindex-updater.ts:24]: I think the default is the a-z order, but even if this deployment script will run before the others, it shouldn't be a problem

Comment (tkeidar) [reply deploy-scripts/arbitrum-test/4-blindex-updater.ts:24]: but anyways that's a NIT - not worth the time investing on that

File: utils/DeployedContractsHelpers.ts [hunk 4]
```diff
@@ -150,10 +150,14 @@ export function getUser(hre: HardhatRuntimeEnvironment): Promise<SignerWithAddre
   return hre.ethers.getNamedSigner("TEST2");
 }
 
-export function getTreasury(hre: HardhatRuntimeEnvironment): Promise<SignerWithAddress> {
+export function getTreasurySigner(hre: HardhatRuntimeEnvironment): Promise<SignerWithAddress> {
   return hre.ethers.getNamedSigner("TREASURY");
 }
 
+export async function getTreasuryAddress(hre: HardhatRuntimeEnvironment): Promise<string> {
+  return hre.network.name === "mainnetFork" ? (await hre.ethers.getNamedSigner("TREASURY")).address : constants.treasuryAddresses[hre.network.name];
```
Comment (tkeidar) [inline utils/DeployedContractsHelpers.ts:158]: lets reuse `getTreasurySigner(hre)` instead of repeating `hre.ethers.getNamedSigner("TREASURY"))`

**Review by tkeidar — APPROVED** [review]
see one last comment and then you can merge

**Pull request merged**